components/storage/blockdevices/COMPONENT_SPIF/SPIFBlockDevice.h

```diff
@@ -0,0 +1,179 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_SPIF_BLOCK_DEVICE_H
+#define MBED_SPIF_BLOCK_DEVICE_H
+
+#include <mbed.h>
+#include "BlockDevice.h"
+
+ 
+/** BlockDevice for SPI based flash devices
+ *  such as the MX25R or SST26F016B
+ *
+ *  @code
+ *  // Here's an example using the MX25R SPI flash device on the K82F
+ *  #include "mbed.h"
+ *  #include "SPIFBlockDevice.h"
+ *  
+ *  // Create flash device on SPI bus with PTE5 as chip select
+ *  SPIFBlockDevice spif(PTE2, PTE4, PTE1, PTE5);
```

Comment:
@yossi2le & @dannybenor : There are many places for various BD's in this PR where these pin names are specified (PTEx - this is K64F specific??). Can we wrap this in an #ifdef to make it target independent? Not all boards have the same pin names.. let's make this more generic for more target platforms..

Reply:
@ashok-rao  this is an example in the remark section for Doxygen.  Do we need to wrap it?

Reply:
@yossi2le : I think I commented on the wrong line.. I meant this one: https://github.com/ARMmbed/mbed-os/pull/7774/files#diff-8a032a4c2b2562763f90905cbe4f39d1R11